driver/src/main/com/mongodb/ServerSessionPool.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright 2017 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb;
+
+import com.mongodb.connection.Cluster;
+import com.mongodb.connection.Connection;
+import com.mongodb.internal.connection.ConcurrentPool;
+import com.mongodb.internal.connection.ConcurrentPool.Prune;
+import com.mongodb.internal.connection.NoOpSessionContext;
+import com.mongodb.internal.validator.NoOpFieldNameValidator;
+import com.mongodb.selector.ReadPreferenceServerSelector;
+import org.bson.BsonArray;
+import org.bson.BsonBinary;
+import org.bson.BsonDocument;
+import org.bson.BsonDocumentWriter;
+import org.bson.UuidRepresentation;
+import org.bson.codecs.BsonDocumentCodec;
+import org.bson.codecs.EncoderContext;
+import org.bson.codecs.UuidCodec;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import static com.mongodb.assertions.Assertions.isTrue;
+import static java.util.concurrent.TimeUnit.MINUTES;
+
+class ServerSessionPool {
+
+    private static final int END_SESSIONS_BATCH_SIZE = 10000;
```

Comment:
Is this size specified? Seems quite large?

Reply:
Yes, it's in the spec.